node/src/effect.rs

```diff
@@ -136,17 +137,30 @@ pub type Effects<Ev> = Multiple<Effect<Ev>>;
 /// size of two items is chosen because one item is the most common use case, and large items are
 /// typically boxed. In the latter case two pointers and one enum variant discriminator is almost
 /// the same size as an empty vec, which is two pointers.
-type Multiple<T> = SmallVec<[T; 2]>;
+// TODO: Move this to a more central spot.
```

Comment:
I'm not sure I agree with this todo for a couple of reasons.  I think it might be better to just leave the type unaliased as it's more obvious to readers then what's going on.  More importantly, I think if this alias becomes more commonly-used, devs will be discouraged from perhaps creating a `SmallVec` with a different default length which might better suit their use-case.

Reply:
It's not quite as common as one might think. Typically there's only three cases: 0, 1 or 2+

If 0-or-1 is a common case, but more is possible (thus ruling out `Option`) and they type would otherwise NOT be boxed, a single element `SmallVec` would be optimal if the type falls into the narrow band of too small to box, too large to hold two of.

Many types will be a bit smaller, so that a 2-item-smallvec is the ideal solution or large enough to warrant a box in any case, making `Vec` optimal.

To sum it up, I believe 80% or more of our cases will be handled by either `Vec` or a two-item `SmallVec` The remaining 20% likely fare decent with a `Vec`.